sql/core/src/test/scala/org/apache/spark/sql/VariantSuite.scala

```diff
@@ -298,4 +300,28 @@ class VariantSuite extends QueryTest with SharedSparkSession {
     }
     assert(ex.getErrorClass == "DATATYPE_MISMATCH.INVALID_ORDERING_TYPE")
   }
+
+  test("variant_explode") {
+    def check(input: String, expected: Seq[Row]): Unit = {
+      withView("v") {
+        Seq(input).toDF("json").createOrReplaceTempView("v")
+        checkAnswer(sql("select pos, key, to_json(value) from v, " +
+          "lateral variant_explode(parse_json(json))"), expected)
+        val expectedOuter = if (expected.isEmpty) Seq(Row(null, null, null)) else expected
+        checkAnswer(sql("select pos, key, to_json(value) from v, " +
+          "lateral variant_explode_outer(parse_json(json))"), expectedOuter)
+      }
+    }
+
+    Seq("true", "false").foreach { codegenEnabled =>
+      withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> codegenEnabled) {
+        check(null, Nil)
+        check("1", Nil)
```

Comment:
@srielau shall we fail if the input variant is not array or struct?

Reply:
oh, this is not a cast, this is exploding the input variant "1". The working example is listed below: `{"a": [1, 2, 3], "b": true}`

Reply:
Yeah, I think that's safest. Otherwise we would have to return an empty result set?

Reply:
As a side note, Snowflake's [flatten](https://docs.snowflake.com/en/sql-reference/functions/flatten) returns an empty result set rather than throwing an error if the variant is not array or struct.